### ISSUE TITLE
linting skips conditional files / sub-tasks

### ISSUE THREAD

**Issue by macwinnie** [ansible/ansible-lint#3285]
##### Summary

Having variables only available in specific inventories but the roles being designed to be “global” leads us to conditional `ansible.builtin.include_tasks` steps.

But with `ansible-lint` we now encountered, that the skipped tasks files also are skipped in regards of the linting process.

Is that expected behaviour for the dev-team? I would not have expected it like that 😅

##### Issue Type

- Expectation clearing question
- Bug Report

##### OS / ENVIRONMENT

Tested on WSL on Windows as well as MacOS, installed via pip in virtual environments. 

##### STEPS TO REPRODUCE

`roles/testing/defaults/main.yml`:

```yaml
username: "appuser"
```

`roles/testing/tasks/main.yml`:

```yaml
---
- name: Do some preflight tests
  ansible.builtin.include_tasks: preflight.yml

- name: Restart service
  ansible.builtin.include_tasks: conditional.yml
  when: restartservice is defined
```

`roles/testing/tasks/preflight.yml`:

```yaml
---
- name: Make sure, user exists
  ansible.builtin.user:
    name: "{{ username }}"
    shell: /bin/sh
    create_home: true
    system: true
```

`roles/testing/tasks/conditional.yml`:

```yaml
- name: Enable and start service
  ansible.builtin.systemd:
    name: "{{ item }}"
    state: started
    enabled: true
    daemon_reload: true
  with_items:
    - "{{ restartservice }}"
    - "{{ undefinedvariablename }}
```

(yeah ... it's a stupid minimal example ... I know ... but it hopefully demonstrates my trouble)

##### Desired Behavior

Linting should not only check files in executable paths but all of them.

##### Actual Behavior

Conditional included tasks is being skipped. Checked by having a dedicated undefined variable `undefinedvariablename` over there.

**Comment by ssbarnea** [comment]
Sorry but what is the the problem? I think that you will need to create a test repository and also provide the ansible-lint command you run with `-v` verbosity, so we can really see what happens.

Did you run `ansible-lint -v` on your repo and discovered that one of the tasks files was ignored or it was something else?

Keep in mind that if you pass it a path that happens to be a role, it might not look at all files within the role.

**Comment by ajinkyau** [comment]
@macwinnie When I tried to reproduce with the example provided, it indeed checks the `conditional.yml` file for errors:
running `ansible-lint roles -vv` has the following lines in the output:
```
DEBUG    Examining roles/my_role/tasks/main.yml of type tasks
DEBUG    Examining roles/my_role/defaults/main.yml of type vars
DEBUG    Examining roles/my_role/tasks/conditional.yml of type tasks
DEBUG    Examining roles/my_role/tasks/preflight.yml of type tasks
```
